### PR TITLE
fix: Fix binding property path for expressions with parentheses and casts

### LINKED ISSUE #8208: `x:Bind` casting not working with `muxc:NavigationViewItem`
### Current behavior

Not sure if just missed scenarios with #4855?

_Both of these scenarios work in UWP_:

```xml
<controls:SwitchPresenter Value="{x:Bind ((muxc:NavigationViewItem)NavView.SelectedItem).Name, Mode=OneWay}">
```

```xml
<controls:SwitchPresenter Value="{x:Bind NavView.SelectedItem.(muxc:NavigationViewItem.Name), Mode=OneWay}">
```

As called out as valid casting methods [in the docs](https://docs.microsoft.com/en-us/windows/uwp/xaml-platform/x-bind-markup-extension#casting)

However, they both cause invalid generated bindings when compiling on WASM/WPF:

```cs
c14.SetBinding(global::Microsoft.Toolkit.Uwp.UI.Controls.SwitchPresenter.ValueProperty, new Windows.UI.Xaml.Data.Binding{ Mode = global::Windows.UI.Xaml.Data.BindingMode.OneWay/* Windows.UI.Xaml.Data.BindingMode/Windows.UI.Xaml.Data.BindingMode, OneWay, Bind/Mode */,  }.BindingApply(___b =>  /*defaultBindModeOneTime ((global::Microsoft.UI.Xaml.Controls.NavigationViewItem)NavView.SelectedItem).Name*/ global::Uno.UI.Xaml.BindingHelper.SetBindingXBindProvider(___b, this, ___ctx => ___ctx is global::MyApp.MainPage ___tctx ? (object)(___tctx.((global::Microsoft.UI.Xaml.Controls.NavigationViewItem)NavView.SelectedItem).Name) : null, null , new [] {"((global::Microsoft.UI.Xaml.Controls.NavigationViewItem)NavView.SelectedItem).Name"})));
```

`error CS1001: Identifier expected`

And for the second one:

```cs
c14.SetBinding(global::Microsoft.Toolkit.Uwp.UI.Controls.SwitchPresenter.ValueProperty, new Windows.UI.Xaml.Data.Binding{ Mode = global::Windows.UI.Xaml.Data.BindingMode.OneWay/* Windows.UI.Xaml.Data.BindingMode/Windows.UI.Xaml.Data.BindingMode, OneWay, Bind/Mode */,  }.BindingApply(___b =>  /*defaultBindModeOneTime NavView.SelectedItem.(global::Microsoft.UI.Xaml.Controls.NavigationViewItem.Name)*/ global::Uno.UI.Xaml.BindingHelper.SetBindingXBindProvider(___b, this, ___ctx => ___ctx is global::MyApp.MainPage ___tctx ? (object)(___tctx.NavView.SelectedItem.(___tctx.global::Microsoft.UI.Xaml.Controls.NavigationViewItem.Name)) : null, null , new [] {"NavView.SelectedItem"})));
```

with `error CS1001: Identifier expected` and `error CS7000: Unexpected use of an aliased name`

### Expected behavior

Able to use a cast to bind to a NavigationViewItem property with x:Bind.

### How to reproduce it (as minimally and precisely as possible)

1. Create an uno project.
2. Add the Toolkit Primitives nugets
3. Create a NavigationView/SwitchPresenter setup:
```xml
     <muxc:NavigationView x:Name="NavView"
                         PaneDisplayMode="Top" IsBackButtonVisible="Collapsed">
        <muxc:NavigationView.MenuItems>
            <muxc:NavigationViewItem x:Name="ItemHelp">
                <muxc:NavigationViewItem.Icon>
                    <FontIcon FontFamily="{ThemeResource SymbolThemeFontFamily}" Glyph="&#xE9CE;" />
                </muxc:NavigationViewItem.Icon>
            </muxc:NavigationViewItem>
            <muxc:NavigationViewItem x:Name="ItemOther" Content="Other" Icon="Calendar"
                                     IsSelected="True"/>
            <muxc:NavigationViewItem x:Name="ItemOther2" Content="Other 2">
                <muxc:NavigationViewItem.Icon>
                    <FontIcon FontFamily="{ThemeResource SymbolThemeFontFamily}" Glyph="&#xF272;" />
                </muxc:NavigationViewItem.Icon>
            </muxc:NavigationViewItem>
        </muxc:NavigationView.MenuItems>
        <controls:SwitchPresenter Value="{x:Bind NavView.SelectedItem.(muxc:NavigationViewItem.Name), Mode=OneWay}">
            <controls:Case Value="ItemHelp">
                <local:HelpPage/>
            </controls:Case>
            <controls:Case Value="ItemOther">
                <local:OtherPage/>
            </controls:Case>
            <controls:Case Value="ItemOther2">
                <local:OtherPage2/>
            </controls:Case>
        </controls:SwitchPresenter>
    </muxc:NavigationView>
```
Use either of the binding setups provided at the top for the SwitchPresenter.
4. Run and see UWP app runs and works.
5. Try compiling for WASM/WPF, get compile error.

### Workaround

Use a regular binding expression:
```xml
<controls:SwitchPresenter Value="{Binding SelectedItem.Name, ElementName=NavView}">
```

### Works on UWP/WinUI

Yes

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia

### NuGet package version(s)

4.2.0-dev.156

### Affected platforms

WebAssembly, Skia (WPF)

### IDE

Visual Studio 2022

### IDE version

17.1.0

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): Part of #8208

## PR Type

What kind of change does this PR introduce?

### <samp>🤖 Generated by Copilot at e26924a</samp>

This pull request improves the x:Bind expression generation and adds a test case for x:Bind with a cast. It refactors the `CSharpBuilder` class to use a separate `StringBuilder` for the property path, and adds a new XAML file and a code-behind file for testing the x:Bind expression with a cast on a `NavigationView` control. It also adds a new test method in the `Given_xBind` test class to verify the functionality.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/Utils/XBindExpressionParser.cs [hunk 1]
```diff
@@ -5,6 +5,7 @@
 using Microsoft.CodeAnalysis.CSharp.Syntax;
 using System;
 using System.Collections.Immutable;
+using System.Diagnostics;
```
Comment (ajpinedam) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/Utils/XBindExpressionParser.cs:8]: Is it required?

File: src/Uno.UI/UI/Xaml/Style/mergedstyles.xaml
Comment (ajpinedam) [inline src/Uno.UI/UI/Xaml/Style/mergedstyles.xaml:1]: Was this file part of the change?

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/Utils/XBindExpressionParser.cs (hunk 1 above)
Comment (Youssef1313) [reply src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/Utils/XBindExpressionParser.cs:8]: Nope. Deleted this using.

File: src/Uno.UI/UI/Xaml/Style/mergedstyles.xaml
Comment (Youssef1313) [reply src/Uno.UI/UI/Xaml/Style/mergedstyles.xaml:1]: No, but these are missing changes on master and this file is auto-generated anyway

**Review by jeromelaban — APPROVED** [review]
🚀

**Pull request merged**